src/APIInterfaces/BackendAPI/Requests/CourseRequests.ts

```diff
@@ -343,7 +347,21 @@ export const postPreviewQuestion = async ({
     webworkQuestionPath,
     problemSeed,
     formData,
+    problemSource,
+    showHints,
+    showSolutions
 }: PreviewQuestionOptions): Promise<AxiosResponse<GetQuestionResponse>> => {
+    if (_.isNil(problemSource) === _.isNil(webworkQuestionPath)) {
+        throw new Error('Either problem source or webwork question path must be defined, not neither, not both');
+    }
+
+    if (!_.isNil(problemSource)) {
+        formData = formData ?? new FormData();
+        formData.set('problemSource', Buffer.from(problemSource).toString('base64'));
```

Comment:
just something to tag for the future: rederly/renderer#7
we will need to re-think this encoding for int'l support